client.go

```diff
@@ -109,30 +113,40 @@ type Client interface {
 
 	// Upsert creates or update a row. A list of fields to update can be
 	// specified. Use All() or nil for all fields.
-	Upsert(context.Context, []string, DomainObject) error
+	// Before calling this method, fill in the DomainObject with ALL
+	// of the primary key fields, along with whatever fields you specify
+	// to update in fieldsToUpdate (or all the fields if you use dosa.All())
+	Upsert(ctx context.Context, fieldsToUpdate []string, objectToUpdate DomainObject) error
 
 	// TODO: Coming in v2.1
 	// MultiUpsert creates or updates multiple rows. A list of fields to
 	// update can be specified. Use All() or nil for all fields.
 	// MultiUpsert(context.Context, []string, ...DomainObject) (MultiResult, error)
 
 	// Remove removes a row by primary key. The passed-in entity should contain
-	// the primary key field values.
-	Remove(context.Context, DomainObject) error
+	// the primary key field values, all other fields are ignored.
+	Remove(ctx context.Context, objectToRemove DomainObject) error
 
 	// TODO: Coming in v2.1
 	// MultiRemove removes multiple rows by primary key. The passed-in entity should
 	// contain the primary key field values.
 	// MultiRemove(context.Context, ...DomainObject) (MultiResult, error)
 
 	// Range fetches entities within a range
-	Range(context.Context, *RangeOp) ([]DomainObject, string, error)
+	// Before calling range, create a RangeOp and fill in the table
+	// along with the partition key information. You will get back
+	// an array of DomainObjects, which will be of the type you requested
+	// in the rangeOp
+	Range(ctx context.Context, rangeOp *RangeOp) ([]DomainObject, string, error)
 
 	// Search fetches entities by fields that have been marked "searchable"
-	Search(context.Context, *SearchOp) ([]DomainObject, string, error)
+	// TODO: Coming in v2.1
+	// Search(ctx context.Context, *SearchOp) ([]DomainObject, string, error)
```

Comment:
is there a reason we named the context parameter but not the searchOp parameter?